### PR TITLE
feat: Add `chat/threads` profile with discord map

### PR DESCRIPTION
**Pull request by martinalbert**
## Description

Adds profile `chat/threads` with use-case `GetThreads` and discord map for this use-case.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: grid/chat/threads/maps/discord.suma [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+  set if (!isNaN(resetTimestamp)) {
+    details.resetTimestamp = resetTimestamp * 1000
+  }
+
+  set if (!isNaN(retryAfter)) {
+    details.retryAfter = retryAfter
+  }
+
+  set if (details.remainingRequests && details.totalRequests) {
```
Comment (janhalama) [inline grid/chat/threads/maps/discord.suma:138]:
```suggestion
  set if (details.remainingRequests !== undefined && details.totalRequests) {
```

**Review by janhalama — APPROVED** [review]
LGTM

File: grid/chat/threads/profile.supr [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+usecase GetThreads {
+  input  {
+    """
+    Workspace
+    Representation of a workspace. Can be guild or workspace, depending on the provider capability.
+    """
+    workspace! string!
+  }
+
+  result {
```
Comment (jnv) [inline grid/chat/threads/profile.supr:21]: This result format doesn't handle pagination.
I know Discord doesn't need it, but it's mentioned in the description. So either add fields for `page` and `nextPage` (which will be always ignored in Discord map), or change the description.

File: grid/chat/threads/profile.supr [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  """
+  Channel
+  Unique identifier of the channel where the thread is located
+  """
+  channel! string!
+
+  """
+  Archive At
+  UNIX timestamp in ms (e.g. 1095379198750) when the thread will be archived
```
Comment (jnv) [inline grid/chat/threads/profile.supr:90]:
```suggestion
  UNIX timestamp in ms (e.g. 1095379198750) when the thread will be archived.
  Some providers limit the lifetime of threads and automatically archive them after some time period of inactivity.
```

File: grid/chat/threads/profile.supr [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+  """
+  Name
+  Name of retrieved thread
+  """
+  name string
+
+  """
+  Members Count
+  Number of members of the retrieved thread
```
Comment (jnv) [inline grid/chat/threads/profile.supr:102]: Is this correct?

```suggestion
  Number of users participating in the thread
```

File: grid/chat/threads/profile.supr [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      workspace = 'not-existing-workspace'
+    }
+
+    error {
+      title = 'workspace_not_found'
+    }
+  }
+}
+
+model Thread {
```
Comment (jnv) [inline grid/chat/threads/profile.supr:69]: If possible, consider adding fields with messages count and the time of the latest message (e.g. `updatedAt` or `lastMessageAt`).

**Review by jnv — CHANGES_REQUESTED** [review]

File: grid/chat/threads/profile.supr (hunk 4 above)
Comment (martinalbert) [reply grid/chat/threads/profile.supr:102]: yep

File: grid/chat/threads/maps/threads.ts [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';
 
 export const getThreadsTest = (
   provider: string,
-  workspace: string,
+  workspaces: string[],
```
Comment (jnv) [inline grid/chat/threads/maps/threads.ts:7]: I'd prefer an object with `valid` and `invalid`, but if you insist on using an array, at least type it as a tuple (I think it's like `[valid: string, invalid: string]`

**Comment by jnv** [comment]
@martinalbert I don't see my comment about pagination being addressed. Do you want to keep it out for now?

**Review by jnv — CHANGES_REQUESTED** [review]
Missing pagination (or ack)

**Comment by martinalbert** [comment]
@jnv if you mean [this](https://github.com/superfaceai/station/pull/201#discussion_r835121206) comment, I already removed description about this usecase being paginated in [here](https://github.com/superfaceai/station/pull/201/commits/cbdb5190d3aff1fe50e358d778db70b28a0b29bf).

**Comment by jnv** [comment]
> if you mean [this](https://github.com/superfaceai/station/pull/201#discussion_r835121206) comment, I already removed description about this usecase being paginated in [here](https://github.com/superfaceai/station/pull/201/commits/cbdb5190d3aff1fe50e358d778db70b28a0b29bf).

OK, my less preferred solution but let's roll with it for now.

**Review by jnv — APPROVED** [review]

**Pull request merged**